### PR TITLE
MULE-15379: Support resources MetadataCache at Mule Runtime level - A…

### PR DESCRIPTION
**Pull request by gsfernandes**
…llow support for sharing persistent object store data between applications

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public PartitionedPersistentObjectStore(MuleContext context) {
   }
 
   @Override
-  public synchronized void open() throws ObjectStoreException {
+  public void open() throws ObjectStoreException {
```
Comment (alegmarra) [inline core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java]: are you sure of this?

File: core/src/main/java/org/mule/runtime/core/internal/store/SharedPartitionedPersistentObjectStore.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.store;
+
+import java.io.File;
+import java.io.Serializable;
+
+public class SharedPartitionedPersistentObjectStore<T extends Serializable> extends PartitionedPersistentObjectStore<T> {
```
Comment (alegmarra) [inline core/src/main/java/org/mule/runtime/core/internal/store/SharedPartitionedPersistentObjectStore.java]: docs

File: core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
   private static final Logger LOGGER = getLogger(PartitionedPersistentObjectStore.class);
   public static final String OBJECT_STORE_DIR = "objectstore";
 
-  private MuleContext muleContext;
+  protected MuleContext muleContext;
```
Comment (alegmarra) [inline core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java]: why?

File: core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java (hunk 1 above)
Comment (gsfernandes) [reply core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java]: Not, my bad. That is not supposed to be changed.

File: core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java (hunk 3 above)
Comment (gsfernandes) [reply core/src/main/java/org/mule/runtime/core/internal/store/PartitionedPersistentObjectStore.java]: Part of PoC, not needed this change. Good catch!

**Pull request merged**